Framework/PythonInterface/plugins/algorithms/SaveReflections.py

```diff
@@ -0,0 +1,259 @@
+SUPPORTED_FORMATS = ["Fullprof", "GSAS", "Jana", "SHELX"]
+NUM_PEAKSWS_COLUMNS = 17
+
+
+def has_modulated_indexing(workspace):
+    """Check if this workspace has more than 3 indicies
+
+    :params: workspace :: the workspace to check
+    :returns: True if the workspace > 3 indicies else False
+    """
+    return workspace.columnCount() > NUM_PEAKSWS_COLUMNS
+
+
+def num_additional_indicies(workspace):
+    """Check if this workspace has more than 3 indicies
+
+    :params: workspace :: the workspace count indicies in
+    :returns: the number of additional indicies present in the workspace
+    """
+    return workspace.columnCount() - NUM_PEAKSWS_COLUMNS
+
+
+def get_additional_index_names(workspace):
+    """Get the names of the additional indicies to export
+
+    :params: workspace :: the workspace to get column names from
+    :returns: the names of any additional columns in the workspace
+    """
+    return ["m{}".format(i+1) for i in range(num_additional_indicies(workspace))]
+
+
+class SaveReflections(PythonAlgorithm):
+
+    def category(self):
+        return "DataHandling"
+
+    def summary(self):
+        return "Saves single crystal reflections to a variety of formats"
+
+    def PyInit(self):
+        """Initilize the algorithms properties"""
+
+        self.declareProperty(FileProperty("Filename", "",
+                                          action=FileAction.Save,
+                                          direction=Direction.Input),
+                             doc="File with the data from a phonon calculation.")
+
+        self.declareProperty(ITableWorkspaceProperty("InputWorkspace", '', Direction.Input),
+                             doc="The name of the peaks worksapce to save")
+
+        self.declareProperty(name="Format",
+                             direction=Direction.Input,
+                             defaultValue="Fullprof",
+                             validator=StringListValidator(SUPPORTED_FORMATS),
+                             doc="The output format to export reflections to")
+
+    def PyExec(self):
+        """Execute the algorithm"""
+        workspace = self.getProperty("InputWorkspace").value
+        output_format = self.getPropertyValue("Format")
+        file_name = self.getPropertyValue("Filename")
+
+        file_writer  = self.choose_format(output_format)
+        file_writer(file_name, workspace)
+
+    def choose_format(self, output_format):
+        """Choose the function to use to write out data for this format
+
+        :param output_format: the format to use to output refelctions as. Options are
+            "Fullprof", "GSAS", "Jana", and "SHELX".
+
+        :returns: file format to use for saving reflections to an ASCII file.
+        """
+        if output_format == "Fullprof":
+            return FullprofFormat()
+        elif output_format == "Jana":
+            return JanaFormat()
+        elif output_format == "GSAS" or output_format == "SHELX":
```

Comment:
Are these likely to diverge in the future? If so then it is a good idea to keep them separate. If not, then perhaps it is better to merge them to `GSAS/SHELX`, just in case someone is wondering why they get the same output for both.

Reply:
As far as I'm aware they probably are not. But my rational was that GSAS and SHELX are two completely different programs that do not necessarily have to support the same format. I guess this way I was just leaving it open in case it ever does change.